stan/math/prim/meta/plain_type.hpp

```diff
@@ -40,13 +42,21 @@ struct eval_return_type {
 template <typename T>
 using eval_return_type_t = typename eval_return_type<T>::type;
 
+namespace internal {
+/**
+ * @brief Used to detect if object has PlainObject
+ */
+template <typename T>
+using plain_type_check_t = typename std::decay_t<T>::PlainObject;
+}  // namespace internal
+
 /**
  * Determines plain (non expression) type associated with \c T. For \c Eigen
  * expression it is a type the expression can be evaluated into.
  * @tparam T type to determine plain type of
  */
 template <typename T>
-struct plain_type<T, require_eigen_t<T>> {
+struct plain_type<T, require_t<is_detected<T, internal::plain_type_check_t>>> {
```

Comment:
What is the reason for this change?

Reply:
`EigenBase` types do not have a `PlainObject` so doing something like `plain_type_t<EigenBase<Whatever>>` will fail

Reply:
Where do you need that? Also your code looks to me like it would fail too. Or am I wrong?

Reply:
It's needed for the tests with checking `is_plain_type`

Actually tho' for `Eigen::EigenBase<Derived>` this would return back the `Eigen::EigenBase<Derived>`, but I think what we want is a specialization for Eigen types without `PlainType` that calls `plain_type_t` on the derived type

Reply:
I added a test to show the usecase I'm talking about. Though it is a weird edge case

Reply:
This is not an actual use case. We are never using `EigenBase` explicitly, so we can safely ignore that. That class is only used internally in Eigen (probably only becouse they don't have our requires).  Even if we did use it in the way it is used in Eigen, we would always have derived type available to use any metaprograms on.